### PR TITLE
Feature : --exclude-failed-response-time.

### PR DESCRIPTION
**Pull request by nmondal**
Issue / Feature:
https://github.com/locustio/locust/issues/2381

### CODE REVIEW

**Comment by nmondal** [comment]
Hello @cyberw  can you please help us with this?

**Comment by cyberw** [comment]
sorry, on holiday

**Comment by cyberw** [comment]
I'm back :)

I think this could be ok. But you'd need to add a test case that verifies the feature. And remove the shortcut name (`-x`), as it could conflict with some other parameter that might be added in the future. I have a few questions as well...

File: locust/env.py [hunk 1]
```diff
@@ -94,6 +92,12 @@ def __init__(
         """
         self.parsed_options = parsed_options
         """Reference to the parsed command line options (used to pre-populate fields in Web UI). When using Locust as a library, this should either be `None` or an object created by `argument_parser.parse_args()`"""
+        self.exclude_failed_response_time = False
+        if self.parsed_options and self.parsed_options.exclude_failed_response_time:
+            self.exclude_failed_response_time = True
+        self.stats = RequestStats(exclude_failed_response_time=self.exclude_failed_response_time)
+        """Reference to RequestStats instance"""
```
Comment (cyberw) [inline locust/env.py:99]: I dont understand this comment.

**Comment by cyberw** [comment]
Hmm.. Iooking more closely at the code it is just not good enough. Too many oddities, and I'd need to spend too much time trying to wrap my head around it. I'm not sure if the old PR worked (https://github.com/locustio/locust/pull/2353), but it was a lot cleaner.

File: locust/env.py (hunk 1 above)
Comment (nmondal) [reply locust/env.py:99]: I did not write it, this is probably a copy paste issue for restructuring.

**Comment by nmondal** [comment]
> Hmm.. Iooking more closely at the code it is just not good enough. Too many oddities in the code, and I'd need to spend too much time trying to wrap my head around it. I'm not sure if the old PR worked (#2353), but it was a lot cleaner.

It won't work. 
Given stats has no clue what to exclude, it won't even know.
It would only work for the case of the option is set to true, and even then it would have all metrics computed wrong.
For example avg, percentile all of the computation would be broken as well as total.
😢 

My first change was exactly the same, and I realized it won't work.

File: locust/env.py [hunk 2]
```diff
@@ -148,7 +152,13 @@ def create_worker_runner(self, master_host: str, master_port: int) -> WorkerRunn
         """
         # Create a new RequestStats with use_response_times_cache set to False to save some memory
         # and CPU cycles, since the response_times_cache is not needed for Worker nodes
-        self.stats = RequestStats(use_response_times_cache=False)
```
Comment (amard33p) [inline locust/env.py:151]: Again, we can simplify it to:
```
self.stats = RequestStats(
    use_response_times_cache=False,
    exclude_failed_response_time=getattr(self.parsed_options, "exclude_failed_response_time", False),
)

**Comment by Alex-TG001** [comment]
Hello, I'm trying to make our final metrics based on what we get from filtering out failed requests. If I changed the logic of the on_requests function like this, would it works?
These codes are in /locust/runners.py
image

**Pull request closed without merging**